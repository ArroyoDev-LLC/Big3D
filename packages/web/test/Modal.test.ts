import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Modal from '../src/components/Modal.vue'

describe('Modal', () => {
  it('should render', () => {
    const wrapper = mount(Modal)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should render header slot', () => {
    const wrapper = mount(Modal, {
      slots: { header: '<div id="header">header</div>' }
    })
    expect(wrapper.find('#header').text()).toBe('header')
  })
  it('should render body slot', () => {
    const wrapper = mount(Modal, {
      slots: { body: '<div id="body">body</div>' }
    })
    expect(wrapper.find('#body').text()).toBe('body')
  })
  it('should render footer slot', () => {
    const wrapper = mount(Modal, {
      slots: { footer: '<div id="footer">footer</div>' }
    })
    expect(wrapper.find('#footer').text()).toBe('footer')
  })
  it('should emit close', () => {
    const wrapper = mount(Modal)
    wrapper.find('i').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('close')
  })
})
