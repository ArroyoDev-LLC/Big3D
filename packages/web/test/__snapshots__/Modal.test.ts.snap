// Vitest Snapshot v1

exports[`Modal > should render 1`] = `
"<div class=\\"modal__backdrop\\">
  <div class=\\"modal\\">
    <header class=\\"modal__header\\"> This is the default title!<i class=\\"pi pi-times p-3 cursor-pointer\\"></i></header>
    <section class=\\"modal__body\\"> I'm the default body!</section>
    <footer class=\\"modal__footer\\">
      <div class=\\"mx-2\\"><button label=\\"Cancel\\" outline=\\"\\"></button></div>
      <div class=\\"mx-2\\"><button label=\\"Save\\"></button></div>
    </footer>
  </div>
</div>"
`;
